packages/hierarchies/src/hierarchies/internal/operators/grouping/BaseClassGrouping.ts

```diff
@@ -29,17 +29,20 @@ export async function createBaseClassGroupsForSingleBaseClass(
 ): Promise<GroupingHandlerResult> {
   const groupedNodes = new Array<ProcessedInstanceHierarchyNode>();
   const ungroupedNodes = new Array<ProcessedInstanceHierarchyNode>();
+  const baseClassFullName = baseECClass.fullName;
   for (const node of nodes) {
     if (
       !node.processingParams?.grouping?.byBaseClasses ||
-      !node.processingParams.grouping.byBaseClasses.fullClassNames.some((className) => className === baseECClass.fullName)
+      !node.processingParams.grouping.byBaseClasses.fullClassNames.some((className) => className === baseClassFullName)
     ) {
       ungroupedNodes.push(node);
       continue;
     }
     const fullCurrentNodeClassName = node.key.instanceKeys[0].className;
 
-    const isCurrentNodeClassOfBase = await baseClassChecker.isECClassOfBaseECClass(fullCurrentNodeClassName, baseECClass);
+    const isCurrentNodeClassOfBase =
+      baseClassChecker.isECClassOfBaseECClassSync(fullCurrentNodeClassName, baseClassFullName) ??
+      (await baseClassChecker.isECClassOfBaseECClass(fullCurrentNodeClassName, baseECClass));
```

Comment:
ugh, this is ugly, I need to think more about this

Reply:
I don't see the point in caching the promise if it would be awaited as soon as it's returned.
Also, I wouldn't say this looks prettier from the consumer's side:
```js
    const baseCheckerResult = baseClassChecker.isECClassOfBaseECClass(fullCurrentNodeClassName, baseECClass);
    const isCurrentNodeClassOfBase = baseCheckerResult instanceof Promise ? await baseCheckerResult : baseCheckerResult;

    if (isCurrentNodeClassOfBase) {
      groupedNodes.push(node);
    } else {
      ungroupedNodes.push(node);
    }
```

Reply:
> I don't see the point in caching the promise if it would be awaited as soon as it's returned.

In case of multiple simultaneous checks for the same class (without `await`), storing a promise while its result is being resolved helps us reuse the same computation instead of creating a new promise for every call.

> Also, I wouldn't say this looks prettier from the consumer's side.

I see a couple of advantages:
1. Because there's only 1 function call to get the result, we have to create the cache key only once.
2. Having two methods for basically the same purpose, but one returning `undefined` in case the value is not cached exposes internal details of `BaseClassChecker` (the fact that it's caching the result). The suggested solution has only one method that always returns a valid result (either value or a promise to it), so consumer doesn't need to worry about making two calls to get it.

Reply:
I see your point now. I've made the suggested changes